src/client.c

```diff
@@ -1,46 +1,175 @@
+static pthread_mutex_t global_mutex = PTHREAD_MUTEX_INITIALIZER;
+static pthread_mutex_t* row_mutexes;
+static pthread_mutex_t* seat_mutexes;
+
+// Inicializa Datos del cliente
 void client_data_init(client_data_t* data, movie_t* movie, int id) {
     data->movie = movie;
     data->id    = id;
 }
 
+// Inicializa el resultado de la reserva.
 void client_res_init(client_res_t* res) {
-    res->success = false; // Por defecto la reserva comienza fallando.
+    res->success = false;
 }
 
-void* client_run(client_t* client) {
-    // De momento el cliente no hace nada.
+// Puntero a función. Se asignará a la función de ejecución que le corresponda.
+void* (*client_run)(void*) = NULL;
+
+// Bloqueo Global
+void* client_global_mutex(void* arg) {
+    client_t* client = (client_t*)arg;
+    movie_t* movie   = client->data.movie;
+    int id           = client->data.id;
+    bool reserved    = false;
+
+    pthread_mutex_lock(&global_mutex); // Bloqueo global
+    int rand_row = rand() % movie->nrows;
+    int rand_col = rand() % movie->ncols;
+    reserved     = movie_reserve_seat(movie, rand_col, rand_row, id);
+    pthread_mutex_unlock(&global_mutex);
+
+    client->res.success = reserved;
     return NULL;
 }
 
-client_t* client_new(movie_t* movie, int id) {
+// Bloqueo por Fila
+void* client_per_row_mutex(void* arg) {
+    client_t* client = (client_t*)arg;
+    movie_t* movie   = client->data.movie;
+    int id           = client->data.id;
+    bool reserved    = false;
+    int row          = id % movie->nrows;
+
+    pthread_mutex_lock(&row_mutexes[row]); // Bloqueo de fila
+    int rand_col = rand() % movie->ncols;
+    reserved     = movie_reserve_seat(movie, rand_col, row, id);
+    pthread_mutex_unlock(&row_mutexes[row]);
+
+    client->res.success = reserved;
+    return NULL;
+}
+
+// Bloqueo por Asiento
+void* client_per_seat_mutex(void* arg) {
+    client_t* client = (client_t*)arg;
+    movie_t* movie   = client->data.movie;
+    int id           = client->data.id;
+    bool reserved    = false;
+
+    int rand_index = rand() % (movie->nrows * movie->ncols);
+    int rand_row   = rand_index / movie->ncols;
+    int rand_col   = rand_index % movie->ncols;
+
+    pthread_mutex_lock(&seat_mutexes[rand_index]); // Bloqueo de asiento
+    reserved = movie_reserve_seat(movie, rand_col, rand_row, id);
+    pthread_mutex_unlock(&seat_mutexes[rand_index]);
+
+    client->res.success = reserved;
+    return NULL;
+}
+
+client_t* client_new(movie_t* movie, int id, int method) {
     client_t* client = (client_t*)malloc(sizeof(client_t));
     if (client == NULL) {
         return NULL;
     }
 
+    switch (method) {
+    case 0:
+        client_run = client_global_mutex;
+        break;
+    case 1:
+        client_run = client_per_row_mutex;
+        break;
+    case 2:
+        client_run = client_per_seat_mutex;
+        break;
+    default:
+        client_run = NULL;
+        free(client);
+        return NULL;
+    }
```

Comment:
Por favor, crea un enum para que sea más fácil entender este switch case.
```c
typedef enum {
    GLOBAL = 0,
    ROW = 1,
    SEAT = 2,
} locking_method;
```

```suggestion
    switch (method) {
    case GLOBAL:
        client_run = client_global_mutex;
        break;
    case ROW:
        client_run = client_per_row_mutex;
        break;
    case SEAT:
        client_run = client_per_seat_mutex;
        break;
    default:
        client_run = NULL;
        free(client);
        return NULL;
    }
```